alicloud/data_source_alicloud_emr_instance_types_test.go

```diff
@@ -0,0 +1,86 @@
+package alicloud
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAlicloudEmrInstanceTypesDataSource_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAlicloudEmrInstanceTypesDataSourceBasicConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAlicloudDataSourceID("data.alicloud_emr_instance_types.foo"),
+					testCheckEmrInstanceTypes("data.alicloud_emr_instance_types.foo"),
+					resource.TestCheckResourceAttrSet("data.alicloud_emr_instance_types.foo", "emr_infos.#"),
+					resource.TestCheckResourceAttrSet("data.alicloud_emr_instance_types.foo", "emr_infos.0.id"),
+					resource.TestCheckResourceAttrSet("data.alicloud_emr_instance_types.foo", "emr_infos.0.zone_id"),
+					resource.TestCheckResourceAttrSet("data.alicloud_emr_instance_types.foo", "main_version"),
+					resource.TestCheckResourceAttrSet("data.alicloud_emr_instance_types.foo", "ids.#"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckEmrInstanceTypes(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		ms := s.RootModule()
+		rs, ok := ms.Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		is := rs.Primary
+		if is == nil {
+			return fmt.Errorf("No primary instance: %s", name)
+		}
+
+		i, err := strconv.Atoi(is.Attributes["emr_infos.#"])
+
+		if err != nil {
+			return fmt.Errorf("convert emr infos length err: %#v", err)
+		}
+
+		if i <= 0 {
+			return fmt.Errorf("emr infos length expected greater than 0 got err: %d", i)
+		}
+
+		instType := is.Attributes["emr_infos.0.id"]
+		if !strings.HasPrefix(instType, "ecs.") {
+			return fmt.Errorf("expected emr_infos.0.id start with [ecs.]; got %s", instType)
+		}
+
+		zoneID := is.Attributes["emr_infos.0.zone_id"]
+		regionID := os.Getenv("ALICLOUD_REGION")
+		if !strings.HasPrefix(zoneID, regionID) {
+			return fmt.Errorf("expected emr_infos.0.zone_id start with [%s]; got %s", regionID, zoneID)
+		}
+
+		mainVersion := is.Attributes["main_version"]
+		if !strings.HasPrefix(mainVersion, "EMR-") {
+			return fmt.Errorf("expected main_version start with [EMR-]; got %s", mainVersion)
+		}
+
+		return nil
+	}
+}
+
+const testAccCheckAlicloudEmrInstanceTypesDataSourceBasicConfig = `
+data "alicloud_emr_instance_types" "foo" {
+	destination_resource = "InstanceType"
+	cluster_type = "HADOOP"
+	instance_charge_type = "PostPaid"
+}
+`
```

Comment:
Please modify the test case with reference to dataSrouce `alicloud_dns_records_test`!